src/utils/taxonomy.ts

```diff
@@ -18,25 +24,90 @@ export type Taxonomy = {
 };
 
 export function getRequestTaxonomy(req: Request): Taxonomy {
+  const taxonomyLevel2 = getTaxonomyLevel2(req);
+
+  let taxonomyLevel1 = TaxonomyLevel1.BLANK;
+  if (isAuthenticationTaxonomy(taxonomyLevel2))
+    taxonomyLevel1 = TaxonomyLevel1.AUTHENTICATION;
+  if (isAccountTaxonomy(taxonomyLevel2))
+    taxonomyLevel1 = TaxonomyLevel1.ACCOUNTS;
+
   return {
-    taxonomyLevel1: TaxonomyLevel1.AUTHENTICATION,
-    taxonomyLevel2: getTaxonomyLevel2(req),
+    taxonomyLevel1,
+    taxonomyLevel2,
   };
 }
 
 function getTaxonomyLevel2(req: Request) {
   let taxonomyLevel2 = TaxonomyLevel2.BLANK;
 
+  if (isSignInTaxonomy(req)) taxonomyLevel2 = TaxonomyLevel2.SIGN_IN;
+  if (isCreateAccountTaxonomy(req))
+    taxonomyLevel2 = TaxonomyLevel2.CREATE_ACCOUNT;
   if (isFeedbackTaxonomy(req)) {
     taxonomyLevel2 = TaxonomyLevel2.FEEDBACK;
     if (isGuidanceTaxonomy(req)) {
       taxonomyLevel2 = TaxonomyLevel2.GUIDANCE;
     }
   }
+  if (isAccountRecoveryTaxonomy(req))
+    taxonomyLevel2 = TaxonomyLevel2.ACCOUNT_RECOVERY;
+  if (isAccountInterventionTaxonomy(req))
+    taxonomyLevel2 = TaxonomyLevel2.ACCOUNT_INTERVENTION;
 
   return taxonomyLevel2;
 }
 
+const isSignInTaxonomy = (req: Request): boolean =>
+  req?.session?.user?.isSignInJourney ||
+  [
+    PATH_NAMES.ENTER_AUTHENTICATOR_APP_CODE,
```

Comment:
I guess similar points here. But maybe this becomes a PR template checklist item for new pages - i.e. make sure the taxonomy middleware reflects the new path.

Reply:
See https://github.com/govuk-one-login/authentication-frontend/pull/2144#discussion_r1793079570